sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -306,13 +306,14 @@ object JdbcUtils extends Logging {
       }
       val metadata = new MetadataBuilder()
       // SPARK-33888
-      // - include scale in metadata for only DECIMAL & NUMERIC
+      // - include scale in metadata for only DECIMAL & NUMERIC as well as ARRAY (for Postgres)
```

Comment:
can we always include the scale metadata if it's present? cc @saikocat

Reply:
We can do that for simplification. But we need to fix the tests for the rest of the dialects. Cos it adds `{"scale": 0}` to all metadata, then existing tests failed (previously metadata didn't get build in getSchema(), so the JSON meta didn't generated)

That's why I choose to set it to only decimal and numeric. The problem also eluded me cos the test for array in Postgresql dialects call the toCalaystType directly instead of going through the code path of using metadata. Sorry on phone so it's hard for me to link the line.

Reply:
Alright, let me elaborate more so you two (cc: @skestle) can decide on which approach to go for. Though I'm kind of favor the current approach of adding data type matching for adding scale metadata cos fixing the failing tests will be more difficult and it makes the JDBCSuite test `"jdbc data source shouldn't have unnecessary metadata in its schema"` test slightly lose its meaning. 

So in order to push "logical_time_type" to metadata, I have to force metadata to be built in the field type as of here:
https://github.com/skestle/spark/commit/0b647fe69cf201b4dcbc0f4dfc0eb504a523571d#diff-c3859e97335ead4b131263565c987d877bea0af3adbd6c5bf2d3716768d2e083R323 whereas previously, metadata can be built  by the dialect or completely ignored (as default)

This will cause 3 tests in `JDBCSuite` to fail cause of schema mismatch (extra `{"scale": 0}` in the metadata always present) [1. `"jdbc API support custom schema"`, 2. `"jdbc API custom schema DDL-like strings."`, 3. `"jdbc data source shouldn't have unnecessary metadata in its schema"`].

Reply:
> whereas previously, metadata can be built by the dialect or completely ignored (as default)

If it was dialect's responsibility to put things into metadata before, I think this PR should put the fix in `PostgresDialect`.

Reply:
But the only way `fieldScale` can make it into the dialect is by the field metadata. So it is a very chicken and egg problem. 

Might have to change the interface somehow to let the ResultSetMetadata to be passed or init-ed to the dialect.